pilot/proxy/envoy/policy.go

```diff
@@ -120,3 +125,128 @@ func applyClusterPolicy(cluster *Cluster,
 		}
 	}
 }
+
+func applyLoadBalancePolicy(cluster *Cluster, policy *routingv2.LoadBalancerSettings) {
+	if policy == nil || cluster.Type == ClusterTypeOriginalDST {
+		return
+	}
+
+	if consistent := policy.GetConsistentHash(); consistent != nil {
+		// cluster.LbType = LbTypeRingHash
+		// TODO: need to set special values in the route config
+		// see: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing.html#ring-hash
+		log.Warn("Consistent hash load balancing type is not currently supported")
+	} else {
+		switch policy.GetSimple() {
+		case routingv2.LoadBalancerSettings_LEAST_CONN:
+			cluster.LbType = LbTypeLeastRequest
+		case routingv2.LoadBalancerSettings_RANDOM:
+			cluster.LbType = LbTypeRandom
+		case routingv2.LoadBalancerSettings_PASSTHROUGH:
+			cluster.LbType = LbTypeOriginalDST // FIXME: double-check
+		default:
+			cluster.LbType = LbTypeRoundRobin
+		}
+	}
+}
+
+func buildOutlierDetection(outlier *routingv2.OutlierDetection) *OutlierDetection {
+	if outlier != nil && outlier.Http != nil {
+		out := &OutlierDetection{
+			ConsecutiveErrors:  5,
+			IntervalMS:         10000, // 10s
+			BaseEjectionTimeMS: 30000, // 30s
+			MaxEjectionPercent: 10,
+		}
+
+		if outlier.Http.BaseEjectionTime != nil {
+			out.BaseEjectionTimeMS = protoDurationToMS(outlier.Http.BaseEjectionTime)
+		}
+		if outlier.Http.ConsecutiveErrors > 0 {
+			out.ConsecutiveErrors = int(outlier.Http.ConsecutiveErrors)
+		}
+		if outlier.Http.Interval != nil {
+			out.IntervalMS = protoDurationToMS(outlier.Http.Interval)
+		}
+		if outlier.Http.MaxEjectionPercent > 0 {
+			out.MaxEjectionPercent = int(outlier.Http.MaxEjectionPercent)
+		}
+
+		return out
+	}
+	return nil
+}
+
+func applyConnectionPool(cluster *Cluster, settings *routingv2.ConnectionPoolSettings) {
+	cluster.MaxRequestsPerConnection = 1024 // TODO: set during cluster construction?
+
+	if settings == nil {
+		return
+	}
+
+	// Envoy's circuit breaker is a combination of its circuit breaker (which is actually a bulk head) and
+	// outlier detection (which is per pod circuit breaker)
+	cluster.CircuitBreaker = &CircuitBreaker{
+		Default: DefaultCBPriority{
+			MaxConnections:     1024, // TODO: what is the Istio default?
```

Comment:
IMO stick to Envoy's defaults here (which is 1024 IIRC).

Reply:
yes, envoy default is 1024, just a possible oversight in the istio proto on what the `istio` default is

Reply:
Just for `MaxConnections`? or all of these? Makes sense for `MaxConnections`, because we don't have an Istio default. Otherwise I'd like to explicitly set the defaults.